js/lib/ui.js

```diff
@@ -11,7 +11,10 @@ elgg.ui.init = function () {
 	// if the user clicks a system message (not a link inside one), make it disappear
 	$(document).on('click', '.elgg-system-messages li', function(e) {
 		if (!$(e.target).is('a')) {
-			$(this).stop().fadeOut('fast');
+			var $this = $(this);
+			$this.clearQueue().slideUp('medium', function () {
```

Comment:
FYI, I went with slideUp because it allows quickly dismissing numerous messages without them moving around unpredictably.